scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -766,27 +766,30 @@ static s32 pick_idle_cpu_from(const struct cpumask *cand_cpumask, s32 prev_cpu,
 	 * partially idle @prev_cpu.
 	 */
 	if (smt_enabled) {
-
-		// if layer is prev_over_idle_core, try prev
-		if (prev_in_cand && layer->prev_over_idle_core &&
-				scx_bpf_test_and_clear_cpu_idle(prev_cpu))
-				return prev_cpu;
-
-		// try prev if layers smt sibling is idle
-		if (prev_in_cand &&
-		    bpf_cpumask_test_cpu(prev_cpu, idle_smtmask) &&
-		    scx_bpf_test_and_clear_cpu_idle(prev_cpu))
+		if (prev_in_cand) {
+			if (layer->prev_over_idle_core) {
+				// try prev if prev_over_idle_core
+				if (scx_bpf_test_and_clear_cpu_idle(prev_cpu))
+					return prev_cpu;
+			} else {
+				// try prev if smt sibling is empty
+				if (bpf_cpumask_test_cpu(prev_cpu, idle_smtmask)
+					&& scx_bpf_test_and_clear_cpu_idle(prev_cpu))
+					return prev_cpu;
+			}
```

Comment:
Don't we still need to do `SCX_PICK_IDLE_CORE` here?

Reply:
yes, thanks